### PR TITLE
chore(linters): Bump golangci-lint to v1.59.1

### PR DESCRIPTION
**Pull request by zak-pawel**
- `golangci-lint` bumped to `v1.59.1`
- Addressing numerous findings caused by updates in several linters:
```
plugins/inputs/cloudwatch/cloudwatch.go:461:3         unparam     (*CloudWatch).aggregateMetrics - result 0 (error) is always nil
plugins/inputs/dovecot/dovecot.go:123:90              unparam     gatherStats - result 0 (error) is always nil
plugins/inputs/http_response/http_response.go:296:28  nolintlint  directive `//nolint:nilerr // error is handled properly` is unused for linter "nilerr"
plugins/inputs/logparser/logparser.go:151:60          unparam     (*LogParserPlugin).tailNewfiles - result 0 (error) is always nil
plugins/inputs/net_response/net_response.go:125:28    nolintlint  directive `//nolint:nilerr // error encoded in result` is unused for linter "nilerr"
plugins/inputs/net_response/net_response.go:132:28    nolintlint  directive `//nolint:nilerr // error encoded in result` is unused for linter "nilerr"
plugins/inputs/net_response/net_response.go:153:28    nolintlint  directive `//nolint:nilerr // error encoded in result` is unused for linter "nilerr"
plugins/inputs/raindrops/raindrops.go:152:13          nolintlint  directive `//nolint:nilerr // nil returned on purpose` is unused for linter "nilerr"
plugins/inputs/sqlserver/sqlserver.go:513:25          unparam     (*SQLServer).getTokenProvider$1 - result 1 (error) is always nil
```
- Matching the configuration file to the example from https://github.com/golangci/golangci-lint/blob/master/.golangci.reference.yml.

### CODE REVIEW

**Comment by zak-pawel** [comment]
I believe that errors from unit tests which appeared on CI are not related to changes in this PR
image

**Review by powersj — APPROVED** [review]
Thanks!

File: plugins/inputs/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -455,13 +456,8 @@ func (c *CloudWatch) gatherMetrics(
...
 
-func (c *CloudWatch) aggregateMetrics(
-	acc telegraf.Accumulator,
-	metricDataResults map[string][]types.MetricDataResult,
-) error {
-	var (
-		grouper = internalMetric.NewSeriesGrouper()
-	)
+func (c *CloudWatch) aggregateMetrics(acc telegraf.Accumulator, metricDataResults map[string][]types.MetricDataResult) {
+	var grouper = internalMetric.NewSeriesGrouper()
```
Comment (srebhan) [inline plugins/inputs/cloudwatch/cloudwatch.go:460]:
```suggestion
	grouper := internalMetric.NewSeriesGrouper()
```

**Review by srebhan — COMMENTED** [review]
Thanks @zak-pawel! Only one cosmetic comment from my side...

File: plugins/inputs/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/cloudwatch/cloudwatch.go]: Done

**Pull request merged**